### DISCUSSION TITLE
Smooth Camera Movement

### DISCUSSION THREAD

**Discussion by ghost** [ikemen-engine/Ikemen-GO#1067] [Feature Request]
### Is your feature request related to a problem? Please describe.

I'd like to propose a smooth camera movement option for gameplay. Unsure how this would be used at the moment, but possibly a stage.def setting you can enable (such as paremeters already there that can control zoom in and out values, maybe something like smoothcamera=1 or 0) or a setting in config.json that can be set to 1 or 0.

Currently, most implementation of smooth/slow camera via ingame scripting (using stuff like helpers) is very hacky (see versions like add004 smooth camera). I've attached a preview of how smooth movement may look (note, this is a heavily modifed add004 camera i made. I managed to make it "manageable" but because of the hackiness of using a helper andsome small things like screenbound weirdness, it still makes the camera jitter in some odd positions).

Having native implementation would be nice.

https://user-images.githubusercontent.com/68257059/226086422-53d2147e-10cb-40bf-b9cb-eb7557aef6d9.mp4

### Describe the solution you'd like

Like the above, possibly this is a setting set in config.json, or stage def

SmoothCamera = 1 or 0

### Search terms

smooth camera movement slow

**Comment by potsmugen** [comment]
The camera tracking speed could definitely be a stage parameter. I think it was just hardcoded in Mugen. At least for the X axis.

**Comment by Lazin3ss** [comment]
I like this camera smoothness idea. I'm working on a camera SCTRL that will greatly expand how camera works (in lots of ways), and there's the idea of having option to add smoothness to the camera in this way. If you want to share how you achieve this effect through helpers, we sould be grateful and it would be a nice input to take in account when developing this idea into the engine.

**Reply by ghost** [reply]
Hi thanks for checkin it out

Here's a test case with base ikemen 0.99 with KFM and mountainside temple and Slow/Smooth Cam code. In data folder it is called SlowCam.zss. For this test case, only 1v1 was in mind when writing the shorter version of the SlowCam zss. I've attached the zip file. 
[SlowCamIkemen099.zip](https://github.com/ikemen-engine/Ikemen-GO/files/11024206/SlowCamIkemen099.zip)
Directions: Run ikemen > 1P vs 2P > KFM vs KFM and MOUNTAINSIDE TEMPLE Stage

Note: I'm not sure if the below is related to this: https://github.com/ikemen-engine/Ikemen-GO/pull/641 
Its character stuttering related to "camera tracking and frame rendering when playing with an odd GameFramerate/GameSpeed" the characters work fine. But it looks like if a Helper is moving the camera, and you set game speed to something like 55, the helper moving the camera will make the root characters stutter if you push and move your opponent. I'll open a seperate issue for that.

**Reply by ghost** [reply]
side related, but opened new issue for relating bug with helper camera movement and stutter when pushing a plyaer, thought id just post it anyway: https://github.com/ikemen-engine/Ikemen-GO/issues/1079
